app/controllers/concerns/filter_parameters.rb

```diff
@@ -25,7 +19,33 @@ def deserialize_array_filter_param(param)
     params[param].split(",") if params.key?(param)
   end
 
+  def remove_previous_parameters(all_parameters)
+    all_parameters.except(*previous_parameters.keys)
+  end
+
+  def merge_previous_parameters(all_parameters)
+    previous_parameters.each do |key, value|
+      next if key =~ /prev_/ && value == "none"
```

Comment:
I find this really hard to parse, you could maybe extract at least the condition to a method that explained what it was doing to make it clearer

Reply:
would it be safer to use `/^prev_/` here instead of `/prev_/`?

Reply:
```suggestion
      next if key.start_with?("prev_") && value == "none"
```